build-release-tools/lib/RepositoryOperator.py

```diff
@@ -326,6 +326,36 @@ def get_lastest_merge_commit_before_date(self, repo_dir, date):
             raise RuntimeError("Unable to get commit id before {date} in directory {repo_dir}"\
                   .format(date=date, repo_dir=repo_dir))
 
+    def get_latest_author_commit_before_date(self, repo_dir, date, author):
+        if repo_dir is None or not os.path.isdir(repo_dir):
+            raise RuntimeError("The repository directory is not a directory")
+        return_code, output, error = self.git.run(['log', '--author='+author, '--format=format:%H', '--before='+date, '-n', '1'], directory=repo_dir)
+
+        if return_code == 0:
+            return output.strip()
+        else:
+            raise RuntimeError("Unable to get commit id of {author} before {date} in directory {repo_dir}"\
+                  .format(author=author, date=date, repo_dir=repo_dir))
+
+
+    def commit1_newer_than_commit2(self, repo_dir, commit1, commit2):
+        '''
+        if commit1 is newer than commit2, return True
```

Comment:
how about returned value when equal?  or just a function  called 'get_newer_commit' and compare the value in the caller, it could avoid comparsion two time both in caller and callee. and the function name is more readable